server.go

```diff
@@ -442,6 +460,10 @@ func (server *Server) loadConfig(configurations configs, globalConfiguration Glo
 							}
 						case types.Wrr:
 							log.Debugf("Creating load-balancer wrr")
+							if stickysession {
+								log.Debugf("... setting to sticky session with cookie named %v", cookiename)
+								rr, _ = roundrobin.New(saveBackend, roundrobin.EnableStickySession(sticky))
+							}
```

Comment:
not needed -- `rr` is defined up at [server.go:419](https://github.com/containous/traefik/pull/404/files#diff-34c6b408d72845d076d47126c29948d1R419) and is the base round robin.  We only need to redefine it in Wrr since the rebalancing round robin uses the base round robin.  So the only time we create a new `rr` object is when we need to pass EnableStickySession to `roundrobin::New`.

For a non-sticky session the fall through `lb = rr` in [server.go:467](https://github.com/containous/traefik/pull/404/files#diff-34c6b408d72845d076d47126c29948d1R467) is sufficient.